meter/tesla.go

```diff
@@ -101,10 +130,38 @@ func NewTesla(uri, usage string) (api.Meter, error) {
 	return decorateTesla(m, totalEnergy, batterySoC), nil
 }
 
+// Login calls login and saves the returned cookie
+func (m *Tesla) Login() error {
+	// username for the powerwall seems to always be customer; email is not required for authentication
+	payload := map[string]interface{}{"password": m.password, "username": "customer"}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	// returns cookie which is saved in the cookie jar
+	resp, err := m.Client.Post(m.uri.String()+teslaLoginURI, "application/json", bytes.NewReader(payloadBytes))
+	if err != nil {
+		return fmt.Errorf("failed to post: %s", err)
```

Comment:
return plain error. it doesn't matter to the user if its due to a post.

Reply:
ok